lib/compat.php

```diff
@@ -518,3 +508,40 @@ function gutenberg_override_reusable_block_post_type_labels() {
 	);
 }
 add_filter( 'post_type_labels_wp_block', 'gutenberg_override_reusable_block_post_type_labels', 10, 0 );
+
+global $current_parsed_block;
+$current_parsed_block = array(
+	'blockName'  => null,
+	'attributes' => null,
+);
+
+/**
+ * Wraps the render_callback of dynamic blocks to keep track
+ * of the current block being rendered via a global variable
+ * called $current_parsed_block.
+ *
+ * This is for get_block_wrapper_attributes to get access
+ * to the runtime data of the block being rendered.
+ *
+ * This shim can be removed when the plugin requires WordPress 5.6.
+ *
+ * @since 9.2.1
+ *
+ * @param array $args Block attributes.
+ * @return array Block attributes.
+ */
+function gutenberg_current_parsed_block_tracking( $args ) {
+	if ( null !== $args['render_callback'] ) {
+		$block_render_callback   = $args['render_callback'];
+		$args['render_callback'] = function( $attributes, $content, $block ) use ( $block_render_callback ) {
```

Comment:
Hi @nosolosw, I've tried running GB 9.2.1 in our CI environment, and I got a test failure that pointed me at this line:

```
ArgumentCountError : Too few arguments to function {closure}(), 2 passed in /home/_redacted_/trunk/wp-includes/class-wp-block-type.php on line 186 and exactly 3 expected
 /home/_redacted_/trunk/wp-content/plugins/gutenberg-core/9.2.1/lib/compat.php:536
 /home/_redacted_/trunk/wp-includes/class-wp-block-type.php:186
 /home/_redacted_/trunk/bin/tests/core/tests/blocks/render.php:325
```

Checking [the callsite of `render_callback`](https://github.com/WordPress/wordpress-develop/blob/fe2053f2c1cff0c416112103988e832687ca3836/src/wp-includes/class-wp-block-type.php#L194) in the [`WP_Block_Type` class](https://github.com/WordPress/wordpress-develop/blob/fe2053f2c1cff0c416112103988e832687ca3836/src/wp-includes/class-wp-block-type.php)' `render` method, I see indeed only 2 arguments passed to `render_callback`.

In the `WP_Block` class' `render` method OTOH, `render_callback` is indeed [invoked](https://github.com/WordPress/wordpress-develop/blob/fe2053f2c1cff0c416112103988e832687ca3836/src/wp-includes/class-wp-block.php#L222) with 3 arguments. I'm not entirely sure which `render` method is called when, but I wanted to flag this, since apparently in our test setup, the wrong one is being called. (Or maybe there shouldn't be two of them?)

cc/ @sirreal @WunderBart @fullofcaffeine

Reply:
Would you mind sharing the full content of the test (which block was it testing and so on)? I'd like to understand that block's attributes and its inner blocks.

Another thing: are you running WordPress 5.6 beta or WordPress 5.5.2?

Finally, there's https://github.com/WordPress/gutenberg/pull/26356 that has improved over this PR, although it hasn't changed the part that is giving you the failure (just moved it to a different place).

Reply:
> Would you mind sharing the full content of the test (which block was it testing and so on)? I'd like to understand that block's attributes and its inner blocks.

Hey @nosolosw, I've edited my original comment to include a pointer to the failing test, not sure you saw:

> Edit: This is the failing test: https://github.com/WordPress/wordpress-develop/blob/fe2053f2c1cff0c416112103988e832687ca3836/tests/phpunit/tests/blocks/render.php#L324. It invokes the `render()` method of an object of class `WP_Block_Type`, so it's quite natural that it fails.

Since this is a Core test, it's a bit surprising that that test wasn't run by GB's own CI :thinking: 

> Another thing: are you running WordPress 5.6 beta or WordPress 5.5.2?

5.5 :slightly_smiling_face: This is a familiar, large-scale, multisite install :wink: 

> Finally, there's #26356 that has improved over this PR, although it hasn't changed the part that is giving you the failure (just moved it to a different place).

:+1:

Reply:
Per DM with @nosolosw, our current hypothesis is that `WP_Block_Type->render()` is basically dead code that's only used by that test, and that we should update that test to remove `WP_Block->render()` instead, and remove `WP_Block_Type->render()` altogether.

Reply:
There's https://github.com/WordPress/gutenberg/pull/26356 to port similar changes to core and it's reporting similar errors. I'm taking a look.